onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -356,6 +479,143 @@ class GpuDFTOperator : public WRL::Base<IMLOperatorKernel>
         return S_OK;
     }
 
+    // { before_dft_axis, axis, after_dft_axis, real_or_complex }
+    std::array<uint32_t, 4> GetReshapedDimensions(gsl::span<const uint32_t> dims, int64_t axis)
+    {
+        std::array<uint32_t, 4> reshapedDims = { 1, 1, 1, 1 };
+        size_t reshapedIndex = 0;
+        for (int i = 0; i < static_cast<int>(dims.size()) - 1; i++)
+        {
+            if (i == axis || i == (axis + 1))
+            {
+                reshapedIndex++;
+            }
+
+            reshapedDims[reshapedIndex] *= dims[i];
+        }
+        return reshapedDims;
+    }
+
+    void PrepareStockhamFFTParams(
+        IMLOperatorKernelContext* context,
+        ID3D12Resource* inputResource,
+        gsl::span<const uint32_t> inputDims,
+        ID3D12Resource* outputResource,
+        gsl::span<const uint32_t> outputDims,
+        uint32_t dftLength,
+        int64_t in_axis,
```

Comment:
>in_axis [](http://example.com/codeflow?start=16&length=7)

```suggestion
inputAxis
```

(nit, casing consistency)

Reply:
Here too:

            auto& zchirp_resource = ...
            auto& b_resource = ...